src/ann_solo/reader.py

```diff
@@ -231,6 +299,324 @@ def get_version(self) -> str:
+            The spectrum in a raw format.
+        identifier : int
+            Incremented identifier of the spectrum in the library.
+
+        Returns
+        -------
+        MsmsSpectrum
+            An MsmsSpectrum object.
+        """
+        # Split raw spectrum in two chunks: metadata & spectrum
+        raw_spectrum_tokens = re.split('Num\s?Peaks:\s?[0-9]+\n',
+                                       raw_spectrum.strip(),
+                                       flags=re.IGNORECASE)
+        spectrum_metadata = raw_spectrum_tokens[0]
+        spectrum = raw_spectrum_tokens[1]
+        # Check if decoy
+        decoy = True if re.search('decoy', spectrum_metadata,
+                                  re.IGNORECASE) else False
+        ##Retrieve peptide & charge
+        peptide_Charge = spectrum_metadata.split('\n', 1)[0].split('/')
+        peptide = peptide_Charge[0].split(' ')[-1].strip()
+        charge = int(peptide_Charge[1].strip())
+        ##Retrieve precurssor mass
+        perMZ = re.search('PrecursorMZ:\s?[0-9]+.[0-9]+', spectrum_metadata,
+                          re.IGNORECASE)
+        if perMZ:
+            perMZ = re.search('[0-9]+.[0-9]+', perMZ.group(0))
+        else:
+            perMZ = re.search('Parent=\s?[0-9]+.[0-9]+', spectrum_metadata,
+                              re.IGNORECASE)
+            perMZ = re.search('[0-9]+.[0-9]+', perMZ.group(0))
+        ##Retrieve MZ & Intensities
+        file = io.StringIO(spectrum)
+        MZ_Intensity_annot = pd.read_csv(file, sep="\t", header=None)
+
+        if MZ_Intensity_annot.shape[1] > 2:
+            annotation = [self._parse_fragment_annotation(mz, annotation)
+                          for mz, annotation in
+                          zip(MZ_Intensity_annot[0], MZ_Intensity_annot[2])]
+        else:
+            annotation = [None] * len(MZ_Intensity_annot[0])
+        spectrum = MsmsSpectrum(identifier, perMZ.group(0), charge,
+                                MZ_Intensity_annot[0].to_numpy(copy=True),
+                                MZ_Intensity_annot[1].to_numpy(copy=True),
+                                annotation,
+                                is_decoy=decoy)
+
+        spectrum.peptide = peptide
+
+        return spectrum
+
+    # For parallel processing
+    def _parse_batch_sptxt_spectra(self, raw_spectra: List[str], ids: List[
+        int]) -> Iterator[MsmsSpectrum]:
+        """
+        Takes a list of  raw spectra data retrieved from an sptxt file and
+        parses it to a structured object of type MsmsSpectrum.
+
+        Parameters
+        ----------
+        raw_spectrum : List[string]
+            The spectrum in a raw format.
+        identifier : List[int]
+            Incremented identifier of the spectrum in the library.
+
+        Returns
+        -------
+        Iterator[MsmsSpectrum]
+            An iterator of spectra in the given library file.
+        """
+        for spectrum in list(Parallel(n_jobs=self._num_threads)(
```

Comment:
**nitpick:** No need to convert to a list.

**todo:** We can just set `n_jobs` to -1 to use all available cores. This would also remove the psutil dependency.